### PR TITLE
Test&input updates

### PR DESCRIPTION
**Pull request by CharlieMorocz**
<h2>Description</h2>
Large refactoring to React components to no longer violate dependency rules, changed the way in which we store intents so that it no longer is reliant on re-rendering the page to display updated intents. Changed navigation between pages to store these intents as constants within the state to preserve them within the front-end. Divided inputs out of UploadPage.js into a pop-up on MainPage.js to only request account info should the user wish to create the prompt on their Voiceflow canvas. Also refactored Jest configuration to accommodate new pages and circumvent errors around styling files.

<h2>Motivation and Context</h2>
Tests were implemented to remove unwanted errors in constantly requesting intents from API as well as reduce burden on the user to input information.

<h2>Bug fixes</h2>
Fixed transition between HelpPage and MainPage so that intents were not lost and MainPage would stop rendering.
Reconfigure Jest to no longer attempt to read css/scss files which was causing issues
Refactored tests to work with new pages.

### CODE REVIEW

File: src/presentation/styles/styleSheets/css/Modal.css [hunk 1]
```diff
@@ -0,0 +1,67 @@
+.backdrop{
+    position:fixed;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background: rgba(0,0,0,0.5);
+    z-index: 1;
+}
+/* .modal{
```
Comment (KareemSalem7) [inline src/presentation/styles/styleSheets/css/Modal.css:10]: delete unneeded commented section

File: src/presentation/styles/styledComponents/StyledHeader.js [hunk 2]
```diff
@@ -114,13 +63,17 @@ function TabGroup({ updatePromptScreen }) {
                 }}
             >
                 {/* Define active tab functionality using onClick to set the active tab be the one selected from the list of types*/}
-                {types.map(type => (
+                {tabNames.map(tabName => (
                     <Tab
-                        key={type}
-                        active={active === type}
-                        onClick={() => setActive(type)}
+                        // here, each tab is mapped to a name in tabNames
```
Comment (KareemSalem7) [inline src/presentation/styles/styledComponents/StyledHeader.js:68]: good comments

File: src/presentation/styles/framerMotionComponents/Modal.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        <AnimatePresence>
+            {showModal && (
+                <motion.div className="backdrop"
+                    variants={animationConstants.backdrop}
+                    initial="hidden"
+                    animate="visible">
+                    <motion.div className="modal" variants={animationConstants.modal}>
+                        <button type="button" className="btn-close btn-close-white btn-block" aria-label="Close" onClick={closer}></button>
+                        <h1>Let's Make Your Chatbot Pop!</h1>
+                        <h2>Almost there! Just input your information below and we'll make your prompt for you!</h2>
```
Comment (rumncoke-ai) [inline src/presentation/styles/framerMotionComponents/Modal.js:27]: I love the "personality" we've choose for guiding the user through the process. By taking a really friendly stance It kind of distracts the users from the fact that were asking for more of their information

File: src/presentation/styles/styledComponents/StyledHeader.js [hunk 4]
```diff
@@ -1,89 +1,38 @@
 // eslint-disable-next-line
-import { BrowserRouter as Router, Link, Route, Routes } from 'react-router-dom';
-import React, { useState, useEffect, useLayoutEffect } from "react";
+import { BrowserRouter as Router, Link, Route, Routes, useLocation } from 'react-router-dom';
```
Comment (KareemSalem7) [inline src/presentation/styles/styledComponents/StyledHeader.js:2]: looks good

**Review by KareemSalem7 — APPROVED** [review]
overall good

File: src/presentation/styles/styledComponents/StyledHeader.js [hunk 5]
```diff
@@ -1,89 +1,38 @@
...
 import NLUlogo from '../../assets/NLULogoTransparent.png';
-import { HoverButtonNoClick } from '../framerMotionComponents/HoverButton';
+import { HoverButtonNoClick, HoverButtonGreyClick } from '../framerMotionComponents/HoverButton';
 import { Tab, StyledDiv } from './StyledConstants';
 
-import { getBestIntents, getDefaultIntents } from '../../../controllers/UserRequests';
-
-// Define the list of types that serve as the selection of tabs on the left
-//export var types = ['Option A', 'Option B', 'Option C'];
+import { tabNames } from '../../pages/MainPage.js';
```
Comment (rumncoke-ai) [inline src/presentation/styles/styledComponents/StyledHeader.js:9]: Good job cleaning up the code, it definitely adds to the readability!

**Review by rumncoke-ai — APPROVED** [review]

File: src/controllers/UserRequests.js [hunk 6]
```diff
@@ -33,151 +85,54 @@ export function sendUserInfo() {
...
-  // apparently this has to be initialized to a certain type, or else an error is caused.
-  // this way, i can change the value in the .then below i think
-  // scratch that, its just the .then scope is screwing up.
-  var intents = ["default intent 1", "default intent 2", "default intent 3"];
 
   //create the fetch request
-  //use threeIntents for now. Hopefully in the future it will be a new getBestIntents controller.
-  intents = await
-  fetch(`http://localhost:8080/api/v1/transcripts/threeIntents`, {method: "GET", headers: headerData})
+  fetch(`http://localhost:8080/api/v1/transcripts/createBlock`, {method: "POST", headers: headerData, body: fullEntry})
```
Comment (molplu) [inline src/controllers/UserRequests.js:132]: We need to change the base of the routes for the backend calls for deployment, here is just one spot.

File: src/presentation/pages/MainPage.js [hunk 7]
```diff
@@ -1,41 +1,62 @@
...
+import Modal from '../styles/framerMotionComponents/Modal.js'
 
 /* define main page function 
 Page responsible for displaying the 3 prompt choices and allowing user to flip through and select one
 Also provides option to view help page and reupload info
 
 */
+
+//This is the type of the voiceflow blocks
+export const tabNames = ["Option A", "Option B"];
```
Comment (molplu) [inline src/presentation/pages/MainPage.js:20]: Are we still using Option A and Option B for tab names? I thought we were using General and Specific.

File: src/presentation/styles/styleSheets/css/Modal.css [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    max-height: 50rem;
+    margin-left: 48rem;
+    margin-bottom: 50rem;
+    padding: 40px 20px;
+    background: white;
+    border-radius: 20px;
+    text-align: center;
+
+} */
+
```
Comment (molplu) [inline src/presentation/styles/styleSheets/css/Modal.css:21]: Maybe add more comments about the modal.

**Review by molplu — APPROVED** [review]
Testing looks good. We could probably add a little bit more documentation to some of the newer pieces of code.

File: src/controllers/UserRequests.js [hunk 9]
```diff
@@ -4,23 +4,75 @@ Javascript file meant for handling fetch requests to the backend.
...
+
+export async function getBestIntents(){
+
+  // retrieve the apiKey and versionID from their respective textboxes
+  var submittedApiKey = document.getElementById("apiKey").value;
+  var submittedApiVersion = document.getElementById("versionID").value;
+
+
+  // we need to encode the api key and version to make sure they are written in characters that can be sent across HTTP
+  var encodedInfo = Buffer.from(submittedApiKey + ":" + submittedApiVersion).toString('base64');
```
Comment (rumncoke-ai) [inline src/controllers/UserRequests.js:29]: Good job on adding more security to the user's information

Comment (MartinCalcaterra) [reply src/controllers/UserRequests.js:29]: This actually isn't security; it is encoding, not encryption. The purpose of encryption is to protect user data. Encoding is just a way to write a string using different characaters. This does not increase security; it just expresses the string in a specific set of characters that can be sent through HTTP headers. The string is decoded back to the actual information in the back end controller.

**Pull request merged**